### PR TITLE
Move on tick from console to room subscription

### PR DESCRIPTION
**Pull request by Flya**
How about this changes for onTick event?

### CODE REVIEW

**Comment by thmsndk** [comment]
@Flya could you explain your idea about how it works? I just woke up and I'm not sure if I understand it :)

We also want to update the tick number based on what tick it says it is in the subscription instead of just incrementing it. 

That way we can detect if a tick is skipped for example. I think it is called time, but the json parser skips it for some reason

Definately some tick architecture we have to think through in regards to multiple kinds of subscriptions from the Web socket ☺

**Comment by thmsndk** [comment]
Check out this issue https://github.com/thmsndk/Screeps3D/issues/110

**Comment by Flya** [comment]
Yes, this changes for #110 issue, i don't found how I can link PR to issue. Great I don't see that we have gameTime in room request, I will use it instead just increment. The main idea is to use room subscription for it, i chose one room subscription if i don't have subscription for tick, and will change it when we unsubscribe from this room.

**Comment by Flya** [comment]
@thmsndk I have updated logic for gam time. Please look on it now. Currently it will update time from room subscription, and fire onTick event once where time changed.

File: Assets/Scripts/Screeps_API/ScreepsSocket.cs [hunk 1]
```diff
@@ -135,10 +135,24 @@ private void FindSubscription(JSONObject json)
...
             }
-
+            if(list[0].str.IndexOf("room:")==0)
+            {
+                findAdUpdateTimeInRoomSubscription(list[1]);
+            }
             _subscriptions[list[0].str].Invoke(list[1]);
         }
 
+        private void findAdUpdateTimeInRoomSubscription(JSONObject json)
```
Comment (thmsndk) [inline Assets/Scripts/Screeps_API/ScreepsSocket.cs]: Spelling mistake xD find**And**UpdateTimeInRoomSubscription

File: Assets/Scripts/Screeps_API/ScreepsSocket.cs [hunk 2]
```diff
@@ -135,10 +135,24 @@ private void FindSubscription(JSONObject json)
...
+        {
+            if(!json.HasField("gameTime") || json["gameTime"] == null)
+            {
+                Debug.LogWarningFormat("Can'r find game time in room subscription: {0}", json);
+                return;
+            }
```
Comment (thmsndk) [inline Assets/Scripts/Screeps_API/ScreepsSocket.cs]: Hmm, not sure we should log this warning all the time. i'm not sure all subscriptions has gameTime. it seems like `roomMap2` does not contain gameTime 🤔 we might want to be specific about what kind of subscriptions we check for this. e.g. !json.list[0].str.Contains("roomMap2") or something like that 🤷  also the initial subscription on a room as far as I understand does not contain the gametime, but i'm not sure :)

**Review by thmsndk — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Screeps_API/ScreepsSocket.cs (hunk 2 above)
Comment (Flya) [reply Assets/Scripts/Screeps_API/ScreepsSocket.cs]: I think I forgot to push latest changes

File: Assets/Scripts/Screeps_API/ScreepsSocket.cs (hunk 1 above)
Comment (Flya) [reply Assets/Scripts/Screeps_API/ScreepsSocket.cs]: Could you pleas check it now.

**Review by thmsndk — APPROVED** [review]
Hmmm, think it seems good now, will test later tonight or tomorrow and then merge :) Have you tried connecting to a server with a user not spawned in? the tick/s in the top corner should be moving along :)

**Comment by Flya** [comment]
Yes, I have tried to connect to botarena.

**Pull request merged**